### DISCUSSION TITLE
Cucumber webdriver IO has failed to taken screenshot on failure step

### DISCUSSION THREAD

**Discussion by RukmaniLTI** [webdriverio/webdriverio#8096] [Community]
Hi,
In my project , i m using Cucumber webdriverIO framework for UI/UX automation. i would like to take screenshot on failure step. but it is not capturing. But the sametime on pass case, it is taking screenshot and getting attached to allure report,

This is code i m using
     afterStep: async function (step, scenario, result, context) {
         if(!result.passed)
           await browser.takeScreenshot();

![image](https://user-images.githubusercontent.com/101386666/157819070-3fb852cd-2532-40a0-b330-56b7f6351874.png)

**Comment by roblesrt** [comment]
You should use the [await browser.saveScreenshot(filepath)](https://webdriver.io/docs/api/browser/saveScreenshot/#usage) .

Also you should set in the wdio config file the reporters settings like:

  reporters: [
    "spec",
    [
      "allure",
      {
        outputDir: "allure-results",
        disableWebdriverStepsReporting: true,
        disableWebdriverScreenshotsReporting: false,
        useCucumberStepReporter: true,
      },
    ],
  ],
